### PR TITLE
Polyfill fetch API, closes #109

### LINKED ISSUE #109: Fetch API Polyfill
It would be great if this service was able to serve a polyfill for the [Fetch API](https://fetch.spec.whatwg.org/) (XHR with Promise interface).

Unfortunately I highly suspect one doesn't yet exist.  Am I wrong, does one already exist?  (Maybe @annevk could confirm?)

If not, then I'm not sure how to proceed...

### PR DESCRIPTION
**Pull request by triblondon**
Early draft.  Don't merge yet.

### CODE REVIEW

**Comment by triblondon** [comment]
A few questions here for @matthew-andrews or @josh:
- Seems like this polyfill depends on `FormData` which is only in IE as of 10. Are you using a polyfill for that?  (eg https://github.com/francois2metz/html5-formdata)
- There's also a soft dependency on `Blob` and `FileReader`.  Is it safe to say these basically aren't polyfillable?  What's the expected impact of using fetch to load blobs in a browser that doesn't support them?
- Testing this is mega hard.  Josh has written a server just to do it, which makes for a very bespoke test harness.  The polyfill service can run Mocha / expect tests.  Do you have any suggestions on how we should test fetch?

**Comment by josh** [comment]
> Seems like this polyfill depends on FormData which is only in IE as of 10. Are you using a polyfill for that?

`FormData` is a soft dependency like `Blob`. If you're host environment doesn't support it, `response.formData()` will just be `undefined`.

I don't know if I would actually recommend using that polyfill since it can't actually provide file upload support. If you need IE 9, you'll probably just not supporting the fancy form upload stuff in that case. Thats basically how github.com deals with it.

> Is it safe to say these basically aren't polyfillable? 

`Blob` _maybe_, though people are reporting that they using some `Blob` polyfill that corrupts utf8 data. So clearly bad ones exist.

> What's the expected impact of using fetch to load blobs in a browser that doesn't support them?

Similar to `FormData`. `response.blob()` will just be `undefined`. And since you have no way to construct a blob, theres no way you could pass a blob to fetch `fetch(body: someBlobWhereWouldItComeFromInIE9)`.

> Testing this is mega hard. Josh has written a server just to do it, which makes for a very bespoke test harness. The polyfill service can run Mocha / expect tests. Do you have any suggestions on how we should test fetch?

Yeah, pretty much anything doing requests will need an actual server. No simple static page. Maybe considering pushing the tests to a Heroku instance just so people can hit the test page publicly. What do you think about that?

File: polyfills/fetch/polyfill.js [hunk 1]
```diff
@@ -0,0 +1,281 @@
+/*
+Copyright (c) 2014 GitHub, Inc.
```
Comment (josh) [inline polyfills/fetch/polyfill.js:2]: I regret not making this public domain to begin with.

File: polyfills/fetch/config.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
+	"aliases": [
+		"es6"
```
Comment (matthew-andrews) [inline polyfills/fetch/config.json]: i don't think `fetch` is ES6?

**Comment by juandopazo** [comment]
You can launch a local server using Grunt in the `test` task.

**Comment by monolithed** [comment]
What are we waiting for, guys? :)

**Comment by juandopazo** [comment]
BTW I recenly tried something like this with grunt-simplemocha and [grunt-run](https://www.npmjs.org/package/grunt-run) and it worked pretty well. You can define a `functional` task as:

``` js
grunt.registerTask('functional', [
  'start:polyfill-service',
  'simplemocha:functional',
  'stop:polyfill-service'
]);

grunt.registerTask('test', ['unit', 'functional']);
```

File: polyfills/fetch/config.json (hunk 2 above)
Comment (monolithed) [inline polyfills/fetch/config.json]: `fetch` is not a part of ES6. 
It's a [host object](https://fetch.spec.whatwg.org/) like WebSoket, FormData and so on.

**Comment by triblondon** [comment]
OK, since this is a straight copy of Josh's polyfill, let's merge this and create a new issue (#267) for testing it.

**Pull request merged**